src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -854,20 +866,24 @@ cdef class _ConcurrentRpcLimiter:
         self._active_rpcs = 0
         self._active_rpcs_condition = asyncio.Condition()
         self._loop = loop
+        self._concurrency_exceeded = False
 
     async def check_before_request_call(self):
         await self._active_rpcs_condition.acquire()
         try:
-            predicate = lambda: self._active_rpcs < self._maximum_concurrent_rpcs
-            await self._active_rpcs_condition.wait_for(predicate)
-            self._active_rpcs += 1
+            if self._active_rpcs >= self._maximum_concurrent_rpcs:
```

Comment:
Now that we've removed the `await`, I think we can get rid of `_active_rpcs_condition` completely. Its only purpose was to enable the `await` here. This will also mean we can get rid of the boilerplate of acquiring and releasing that condition.

This also means that this method no longer needs to be `async`.